src/server/main_service.cc

```diff
@@ -1318,7 +1318,8 @@ bool Service::InvokeCmd(const CommandId* cid, CmdArgList tail_args, SinkReplyBui
     DispatchMonitor(cntx, cid, tail_args);
   }
 
-  ServerState::tlocal()->RecordCmd();
+  const bool is_main = cntx->conn() ? cntx->conn()->IsMain() : false;
```

Comment:
Cn `cntx->conn()` be null here ? If so maybe we can store the `IsMain` in `ConnectionContext` instead ?

Reply:
I saw this was null during a unit test which runs the `populate` debug command, because it called `InvokeCmd` with a locally created context which had owner set to nullptr https://github.com/dragonflydb/dragonfly/blob/main/src/server/debugcmd.cc#L196

Reply:
yep this is a `stub context` and we have a bug. So, why don't we use the context to check if `it's main` or not ?